src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs

```diff
@@ -0,0 +1,221 @@
+            }
+        }
+
+        private void OnClassificationFormatMappingChanged(object sender, EventArgs e)
+        {
+            // this method should only run on UI thread as we do WPF here.
+            Contract.ThrowIfFalse(TextView.VisualElement.Dispatcher.CheckAccess());
+
+            if (AdornmentLayer is not null)
+            {
+                foreach (var element in AdornmentLayer.Elements)
+                {
+                    var tag = (InlineDiagnosticsTag)element.Tag;
+                    var classificationType = _classificationRegistryService.GetClassificationType(InlineDiagnosticsTag.GetClassificationId(tag.ErrorType));
+                    var format = GetFormat(classificationType);
+                    InlineDiagnosticsTag.UpdateColor(format, element.Adornment);
+                }
+            }
+        }
+
+        private TextFormattingRunProperties GetFormat(IClassificationType classificationType)
+        {
+            return _formatMap.GetTextProperties(classificationType);
+        }
+
+        /// <summary>
+        /// Get the spans located on each line so that it can only display the first one that appears on the line
+        /// </summary>
+        private IDictionary<int, IMappingTagSpan<InlineDiagnosticsTag>> GetSpansOnEachLine(NormalizedSnapshotSpanCollection changedSpanCollection)
+        {
+            var map = new Dictionary<int, IMappingTagSpan<InlineDiagnosticsTag>>();
+            var viewLines = TextView.TextViewLines;
+
+            foreach (var changedSpan in changedSpanCollection)
+            {
+                if (!viewLines.IntersectsBufferSpan(changedSpan))
+                {
+                    continue;
+                }
+
+                var tagSpans = TagAggregator.GetTags(changedSpan);
+                foreach (var tagMappingSpan in tagSpans)
+                {
+                    if (!ShouldDrawTag(changedSpan, tagMappingSpan))
+                    {
+                        continue;
+                    }
+
+                    var mappedPoint = GetMappedPoint(changedSpan, tagMappingSpan);
+
+                    // mappedPoint is known to not be null here because it is checked in the ShouldDrawTag method call.
+                    var lineNum = mappedPoint!.Value.GetContainingLine().LineNumber;
+
+                    // If the line does not have an associated tagMappingSpan and changedSpan, then add the first one.
+                    if (!map.TryGetValue(lineNum, out var value))
+                    {
+                        map.Add(lineNum, tagMappingSpan);
+                    }
+
+                    // If the map has a value and the value is not a syntax error, then rewrite the value in the map since
+                    // each line with errors will be ordered by importance
+                    if (value is not null && value.Tag.ErrorType is not PredefinedErrorTypeNames.SyntaxError)
+                    {
```

Comment:
```suggestion
                    }
                    else if (value is not null && value.Tag.ErrorType is not PredefinedErrorTypeNames.SyntaxError)
                    {
                        // If the map has a value and the value is not a syntax error, then rewrite the value in the map since
                        // each line with errors will be ordered by importance
```